### ISSUE TITLE
Missing file ./lib/parser/tokenizer-event-handlers

### ISSUE THREAD

**Issue by tpitale** [glimmerjs/glimmer-vm#459]
It appears that `index.d.ts` tries to `export { preprocess } from './lib/parser/tokenizer-event-handlers';` but the `parser` folder in `@glimmer/syntax/types/lib` is not present.

Resulting error:

```
ERROR in node_modules/@glimmer/syntax/dist/types/index.d.ts
(1,28): error TS2307: Cannot find module './lib/parser/tokenizer-event-handlers'.
```

Using (from yarn.lock file):

```
"@glimmer/compiler@^0.23.0-alpha.11":
  version "0.23.0-alpha.11"
  resolved "https://registry.yarnpkg.com/@glimmer/compiler/-/compiler-0.23.0-alpha.11.tgz#2f2ea9a4e20ed7f080f650a04f1d5c0385a97739"
  dependencies:
    "@glimmer/interfaces" "^0.23.0-alpha.11"
    "@glimmer/syntax" "^0.23.0-alpha.11"
    "@glimmer/util" "^0.23.0-alpha.11"
    "@glimmer/wire-format" "^0.23.0-alpha.11"
    simple-html-tokenizer "^0.3.0"
```

**Comment by mlunoe** [comment]
Digging a bit deeper I can see that these `.d.ts` files are actually missing, causing the error to occur:
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.d.ts
packages/@glimmer/syntax/lib/parser.d.ts

It seems that there is actually a problem with the build for this project as it does not produce those files when I run the build. However, if I build the `.ts` files directly then they are produced (though throughout the project there seems to be _a lot_ of errors thrown from building it):
```
glimmer-vm mlunoe ((v0.24.0-beta.4) *)$ ./node_modules/.bin/tsc -d packages/@glimmer/syntax/lib/parser.ts 
packages/@glimmer/syntax/lib/parser.ts(8,24): error TS2307: Cannot find module '@glimmer/interfaces'.
packages/@glimmer/syntax/lib/parser.ts(9,32): error TS2307: Cannot find module '@glimmer/util'.
packages/@glimmer/syntax/lib/parser.ts(48,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(52,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(58,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(64,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(70,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(76,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
glimmer-vm mlunoe ((v0.24.0-beta.4) *)$ ./node_modules/.bin/tsc -d packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts 
packages/@glimmer/syntax/lib/builders.ts(2,24): error TS2307: Cannot find module '@glimmer/interfaces'.
packages/@glimmer/syntax/lib/parser.ts(8,24): error TS2307: Cannot find module '@glimmer/interfaces'.
packages/@glimmer/syntax/lib/parser.ts(9,32): error TS2307: Cannot find module '@glimmer/util'.
packages/@glimmer/syntax/lib/parser.ts(48,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(52,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(58,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(64,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(70,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/parser.ts(76,7): error TS1056: Accessors are only available when targeting ECMAScript 5 and higher.
packages/@glimmer/syntax/lib/utils.ts(2,24): error TS2307: Cannot find module '@glimmer/interfaces'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(5,24): error TS2307: Cannot find module '@glimmer/interfaces'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(82,23): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(86,19): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(96,19): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(100,19): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(106,19): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(133,19): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(141,23): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts(152,79): error TS2341: Property 'state' is private and only accessible within class 'EventedTokenizer'.
```

cc. @lennyburdette, @eventualbuddha

**Comment by mlunoe** [comment]
Not entirely sure, but this seems to change the build quite a bit. I tried looking into fixing this but gave up. Maybe somebody from the team could help out with explaining where and why some files' `.d.ts` counterparts are not being generated?

**Comment by mlunoe** [comment]
Running `./node_modules/.bin/tsc -p tsconfig.json` (i.e. `npm run problems`, but without the `--noEmit` option) will actually produce all the files, but I guess there is a reason for the build being configured to move everything to dist (that is, except the missing declaration files).

**Comment by mlunoe** [comment]
So, I was able to get things working by building the project and manually moving the manually produced `*.d.ts` files into the right spot (and the other missing files too, if needed):
`node_modules/@glimmer/syntax/dist/types/lib/parser.d.ts` and `node_modules/@glimmer/syntax/dist/types/lib/parser/handlebars-node-visitors.d.ts`.

It seems this is a problem with the build. There must be some place in the [build-packages.js](https://github.com/glimmerjs/glimmer-vm/blob/master/build/broccoli/build-packages.js) file that excludes them somehow, or for some reason doesn't produce them in the first place.

At this point I am unable to get further to actually fix the build, otherwise I would gladly put up PR with the fix, but I do need some help from the maintainers to get further.

**Comment by rwjblue** [comment]
There is an open PR that @tomdale is working that unblocks folks for now, but the underlying issue is with the build (as you have identified). 

Also, an updated version of broccoli-typescript-compiler is forthcoming that should provide much better support for normal node_modules resolution (and eliminate a number of the errors from our build output).

**Comment by tomdale** [comment]
This should be fixed by glimmerjs/glimmer-vm#491, release coming soon…

**Comment by tomdale** [comment]
@mlunoe This should be fixed in `@glimmer/syntax@0.24.0-beta.5`, please let me know if you're still having issues with that release!

**Comment by mlunoe** [comment]
Thank you @tomdale that does fix my situation! I am looking forward to the update of `broccoli-typescript-compiler`. That would make it much easier to debug/navigate what is happening in the build!
